### PR TITLE
Addition of Squid bespoke access log content pack (Linux client and Stroom content)

### PR DESCRIPTION
**Pull request by burnalting**
A Linux Squid access log content pack containing both the Squid configuration details, Linux scripts to acquire, enrich, reformat and post to a Stroom service as well as the Stroom content pack that will normalise the access logs into the Stroom

### CODE REVIEW

File: source/squidplus-proxy/clientArtefacts/README [hunk 1]
```diff
@@ -0,0 +1,103 @@
+
+Release 1.0.0	- 20170630 Burn Alting - burn@swtf.dyndns.org
+  - Initial Release
+
+Synopsis:
+========
+The script squid_stroom_feeder.sh and it's supporting perl script, squidplusXML.pl, is designed to run from a crontab entry
+that periodically collects and enriches access events from an appropriately configured single Squid Proxy service and posts the enriched events
+to a Stroom audit repository (web service). It is expected that
```
Comment (at055612) [inline source/squidplus-proxy/clientArtefacts/README]: 'to Stroom' or 'to an instance of Stroom'

File: source/squidplus-proxy/clientArtefacts/README [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  - your squid deployment has been set up to generate SquidPlus format squid access logs (see later)
+  - the Stroom Audit System has been configured to accept streams of events in the SQUID-PLUS-XML-V1.0-EVENTS feed.
```
Comment (at055612) [inline source/squidplus-proxy/clientArtefacts/README]: Stroom Audit System => Stroom

File: source/squidplus-proxy/clientArtefacts/README [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Prerequisites:
+=============
+  - Squid configured appropriately (see later)
+  - requires the bind-utils, coreutils, curl, net-tools, gzip and sed packages. Also requires both perl and perl(XML::Simple) packages as well
+
+Capability Workflow:
+===================
+The workflow is such that, every 10 minutes cron starts a script which
+  - delays randomly between 7 and 580 seconds before collecting audit data in order to balance network load. One does want many Linux
```
Comment (at055612) [inline source/squidplus-proxy/clientArtefacts/README]: does => doesn't ?

File: source/squidplus-proxy/stroomContent/SQUID-PLUS-XML.Folder.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.1" encoding="UTF-8"?>
```
Comment (at055612) [inline source/squidplus-proxy/stroomContent/SQUID-PLUS-XML.Folder.xml]: We need to agree a convention for the folder structure in stroom.  I think the system specific content folders should be bundled into one or more parent folders to keep things tidy.

File: source/squidplus-proxy/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+* Dependancies
+
+This content is dependant on the `Event Data (XML)` Pipeline from the [`template-pipelines`](../template-pipelines/README.md) Content Pack.
+
+## Client Contents
+
+The client artefacts are
+
+* **README** `Document`
```
Comment (burnalting) [inline source/squidplus-proxy/README.md]: But if we tag the Stroom elements why not be consistent and tag the client elements?

Comment (at055612) [inline source/squidplus-proxy/README.md]: Happy to leave them in

File: source/squidplus-proxy/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+## Summary
+
+The _squidplus-proxy_ Content Pack provides both client artefacts that acquire then post Squid access logs to a Stroom instance and Stroom content
+artefacts that will normalise the Squid access logs into the Stroom [`event-logging-schema`](https://github.com/gchq/event-logging-schema) format.
+
+This package does not use the standard Squid acces log format, but a bespoke format called _squidplus_. This format records additional information such
+as port numbers, complete request and response headers, additional status information, data transfer sizes and next hop host information.
+
+Client deployment information can be found in the supplied README file.
```
Comment (stroomdev10) [inline source/squidplus-proxy/README.md]: also spelling

+This package does not use the standard Squid acces log format, but a bespoke format called _squidplus_. This format records additional information such

should be "access"

File: source/squidplus-proxy/clientArtefacts/squid_stroom_feeder.sh [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+    ;;
+  \?)
+    echo "$0: Invalid option -$OPTARG"
+    echo $Usage
+    exit 1
+    ;;
+ esac
+done
+
+# FEED_NAME         - Name of feed (asssigned by Audit Authority)
```
Comment (stroomdev10) [inline source/squidplus-proxy/clientArtefacts/squid_stroom_feeder.sh]: This is superfluous, it is duplicated further down

File: source/squidplus-proxy/clientArtefacts/squid_stroom_feeder.sh [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+# Samples are
+#   Basic extraction from a rpm package
+#       VERSION=`rpm -q httpd`
+#UBUNTU VERSION=`dpkg --status squid | awk '{if ($1 == "Version:") print $2;}'`
+VERSION=`rpm -q squid`
+
+
+# FEED_NAME         - Name of Stroom feed (asssigned by Audit Authority)
+#
+# We work out which feed we want based on the Linux audit version running and possible
```
Comment (stroomdev10) [inline source/squidplus-proxy/clientArtefacts/squid_stroom_feeder.sh]: Comment doesn't make sense here

**Comment by burnalting** [comment]
Closing PR as it's against too old a branch. Will re-issue.

**Pull request closed without merging**